scripts/filter_index.py

```diff
@@ -42,8 +43,8 @@
 
 def parse_arguments():
     parser = ArgumentParser(description=__doc__)
-    parser.add_argument('input_dir', help='the input directory.')
-    parser.add_argument('output_dir', help='the output directory.')
+    parser.add_argument('--input_dir', '-i', help='the input directory.')
```

Comment:
Aren't these required arguments? I would say they should stay positional then.

Reply:
You are right, they are required.

But I got rid off positional arguments and instead adopted unified shortcodes (always -i and -o, and always -P for process if that is applicable) for these arguments, so the manager app can put together the proper arguments for each script easily.

But I will add the required=True flag, that's a good idea.